src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -171,14 +180,12 @@ async Task InnerProcessMessages()
 
         Task ReceiveMessage()
         {
-            return TaskEx.Run(async state =>
+            return Task.Run(async () =>
```

Comment:
this seems to be a performance optimization to avoid variable capturing. Has this been intentionally removed?

Reply:
ah, just aligned with some of the other seams. I'll revert and add a comment

Reply:
Fixed

Reply:
or actually, does this work as intended with me inlining it?

Reply:
The trick is to access the state and use that as a host to avoid the closure. So no most likely what you did there doesn't work as before

Reply:
Got it!  I've added the state access back and some comments. Does it look right?